flink-connectors/flink-connector-hbase-2.2/src/main/java/org/apache/flink/connector/hbase2/HBase2DynamicTableFactory.java

```diff
@@ -136,4 +133,20 @@ public String factoryIdentifier() {
         set.add(LOOKUP_MAX_RETRIES);
         return set;
     }
+
+    @Override
+    public Set<ConfigOption<?>> forwardOptions() {
```

Comment:
ConnectorOptions will be used to generate the docs above via hugo. It might be better to extend the forward info there, for HBase i.e. HBaseConnectorOptions. This is a common issue that has impact on all connectors.

Reply:
But in case of table factories, the docs generator will use the `Factory` interface anyway. We've paved the idea of extending `ConfigOption`, but it was way too complicated and impactful, hence we ended up with this method

Reply:
Which doc generator and Factory interface do you refer to? Afaik, the `ConfigOptionsDocGenerator` only works with `ConfigOption`.

Reply:
Exactly, but we still need to adapt it to our `Factory` interface.

Reply:
Does it make sense to adapt it in this PR too? Otherwise the `ConfigOptionsDocGenerator` will be broken and hard to track the reason in the future.

Reply:
I think we should do it in another PR, as it's not trivial, because we also need to figure out how to include the generated docs within the docs engine

Reply:
Thanks for the info. In this case, this PR will break thing and it looks like fixing it will take long time. I would suggest letting @chesnay be aware of this.

Reply:
How is this PR going to break things? We don't have a working `ConfigOptionsDocGenerator` for table factories now.

Reply:
Now I got your point. We had content-out-of-synch issue previously and there is a [ticket](https://issues.apache.org/jira/browse/FLINK-25506) for doing it, at least for HBase